src/liballoc/collections/vec_deque.rs

```diff
@@ -1920,6 +1920,40 @@ impl<T: Clone> VecDeque<T> {
             self.truncate(new_len);
         }
     }
+
+    /// Modifies the `VecDeque` in-place so that `len()` is equal to `new_len`,
+    /// either by removing excess elements from the back or by appending
+    /// elements generated by calling `generator` to the back.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(vec_resize_with)]
+    ///
+    /// use std::collections::VecDeque;
+    ///
+    /// let mut buf = VecDeque::new();
+    /// buf.push_back(5);
+    /// buf.push_back(10);
+    /// buf.push_back(15);
+    /// assert_eq!(buf, [5, 10, 15]);
+    ///
+    /// buf.resize_with(2, || unreachable!());
+    /// assert_eq!(buf, [5, 10]);
+    ///
+    /// buf.resize_with(5, Default::default);
+    /// assert_eq!(buf, [5, 10, 0, 0, 0]);
```

Comment:
Please include a test with a closure that doesn't return the same result every time, such as `|| { x += 1; x }` (given a preceding `let mut x = 0;`).

Other than that, LGTM.

Reply:
Makes sense!  Test added.